nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractEmailAttachments.java

```diff
@@ -0,0 +1,209 @@
+
+        session.read(originalFlowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream rawIn) throws IOException {
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Properties props = new Properties();
+                        Session mailSession = Session.getDefaultInstance(props, null);
+                        MimeMessage originalMessage = new MimeMessage(mailSession, in);
+                        MimeMessageParser parser = new MimeMessageParser(originalMessage).parse();
+                        // RFC-2822 determines that a message must have a "From:" header
+                        // if a message lacks the field, it is flagged as invalid
+                        Address[] from = originalMessage.getFrom();
+                        Date sentDate = originalMessage.getSentDate();
+                        if (from == null || sentDate == null ) {
+                            // Throws MessageException due to lack of minimum required headers
+                            throw new MessagingException("Message failed RFC2822 validation");
+                        }
+                        if (parser.hasAttachments()) {
+                            final String originalFlowFileName = originalFlowFile.getAttribute(CoreAttributes.FILENAME.key());
+                            for (final DataSource data : parser.getAttachmentList()) {
+                                try {
+                                FlowFile split = session.create(originalFlowFile);
+                                final Map<String, String> attributes = new HashMap<>();
+                                if (StringUtils.isNotBlank(data.getName())) {
+                                    attributes.put(CoreAttributes.FILENAME.key(), data.getName());
+                                }
+                                if (StringUtils.isNotBlank(data.getContentType())) {
+                                    attributes.put(CoreAttributes.MIME_TYPE.key(), data.getContentType());
+                                }
+                                String parentUuid = originalFlowFile.getAttribute(CoreAttributes.UUID.key());
+                                attributes.put(ATTACHMENT_ORIGINAL_UUID, parentUuid);
+                                attributes.put(ATTACHMENT_ORIGINAL_FILENAME, originalFlowFileName);
+                                split = session.append(split, new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        IOUtils.copy(data.getInputStream(),out);
+                                    }
+                                });
+                                split = session.putAllAttributes(split, attributes);
+                                originalFlowFilesList.add(originalFlowFile);
```

Comment:
How originalFlowFile gets handled needs to be revisited. Currently in every iteration of the for loop it is added to the the originalFlowFilesList.